astrovascpy/bloodflow.py

```diff
@@ -51,6 +51,11 @@
 
 
 L = logging.getLogger(__name__)
+warning_msg = "The program can be slow and the result not very accurate.\
```

Comment:
3 things:
- `warning_msg` is constant. It should be upper-cased.
- the variable is only used in function `_solve_linear`, so the constant should be defined within the scope of the function
- You may use the string formatting operator handled by the [logging methods](string formatting operator) instead of building the message the way you are doing. You can also use `textrap.dedent` utility function defined in the standard library to make the constant definition prettier. For instance:
   ```python
   WARNING_MSG = textwrap.dedent("""\
   	Number of nodes = %(n_nodes)s.
       The program can be slow and the result not very accurate.
       It is recommended to use PETSc for a graph with more than 2e6 nodes and
       SciPy for a graph with less than 2e6 nodes.
       The default solver can be selected in setup.sh or by setting the environment
       variable BACKEND_SOLVER_BFS to 'scipy' or 'petsc'.
   """)
   L.warning(WARNING_MSG, {"n_nodes": n_nodes})
   ```

Reply:
That looks nice. I didn't know the dedent function. Thank you.